packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js

```diff
@@ -3,51 +3,99 @@ import { shape, string, bool, func } from 'prop-types';
 import { RadioGroup } from 'informed';
 
 import Radio from '../../RadioGroup/radio';
-import CreditCard from './creditCard';
 import { mergeClasses } from '../../../classify';
 
+import { usePaymentMethods } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentMethods';
+import paymentMethodOperations from './paymentMethods.gql';
 import defaultClasses from './paymentMethods.css';
 
+import CreditCard from './creditCard';
+import FreePaymentMethod from './PaymentMethods/FreePaymentMethod/freePaymentMethod';
+import CheckmoMethod from './PaymentMethods/CheckmoMethod/checkmoMethod';
+
 const PaymentMethods = props => {
     const {
         classes: propClasses,
         reviewOrderButtonClicked,
         selectedPaymentMethod,
+        setDoneEditing,
         onPaymentSuccess,
         onPaymentError,
         resetReviewOrderButtonClicked
     } = props;
 
     const classes = mergeClasses(defaultClasses, propClasses);
 
-    const creditCard =
-        selectedPaymentMethod === 'braintree' ? (
-            <CreditCard
+    const talonProps = usePaymentMethods({
+        ...paymentMethodOperations
+    });
+
+    const {
+        availablePaymentMethods,
+        initialSelectedMethod,
+        isLoading
+    } = talonProps;
+
+    if (isLoading) {
+        return <div>Loading methods...</div>;
+    }
+
+    const COMPONENTS = {
+        free: (
+            <FreePaymentMethod
+                onSubmit={onPaymentSuccess}
+                setDoneEditing={setDoneEditing}
                 shouldSubmit={reviewOrderButtonClicked}
+            />
+        ),
+        braintree: (
+            <CreditCard
                 brainTreeDropinContainerId={
                     'checkout-page-braintree-dropin-container'
                 }
                 onPaymentSuccess={onPaymentSuccess}
                 onPaymentError={onPaymentError}
                 resetShouldSubmit={resetReviewOrderButtonClicked}
+                setDoneEditing={setDoneEditing}
+                shouldSubmit={reviewOrderButtonClicked}
             />
-        ) : null;
+        ),
+        checkmo: (
+            <CheckmoMethod
+                onSubmit={onPaymentSuccess}
+                setDoneEditing={setDoneEditing}
+                shouldSubmit={reviewOrderButtonClicked}
+            />
+        )
+    };
+
+    const radios = availablePaymentMethods.map(({ code, title }) => {
+        // Always enable "free".
+        const isSelected = selectedPaymentMethod === code || code === 'free';
```

Comment:
This is weird. We don't render `component` until the radio is clicked, so the component logic that tells the UI "I'm done!" is only invoked if the component is clicked by a user (or if we automatically select it such as in the case of `free`.

How do I get out of that?